src/main/java/org/edgexfoundry/serviceactivator/MQTTOutboundServiceActivator.java

```diff
@@ -63,5 +68,24 @@ public String mqttOutbound(Message<?> msg) {
 		}
 		return null;
 	}
+	
+	private void addPath(Addressable addressable, String deviceId) {
+		// if path contains "/" pre- or append path to topic, otherwise
+		// if path contains "\[any_string]" pre- or append dynamically device id to topic
+		// TODO - read "\[key]" from addressable and pre- or append its value to topic
+		String path = addressable.getPath();
+		if (path != null && !path.equals("")) {
+			String topic = addressable.getTopic();
+			if (path.charAt(0) == '/') {
+				addressable.setTopic(topic + path);
+			} else if (path.charAt(path.length() - 1) == '/') {
+				addressable.setTopic(path + topic);
+			} else if (path.charAt(0) == '\\') {
+				addressable.setTopic(topic + "/" + deviceId);
+			} else if (path.charAt(path.length() - 1) == '\\') {
```

Comment:
Not a big fan of overloading the path like this, but even if we accept that then this needs to change in my opinion. This case:

> path = '\\';

Might look like it will prepend the device ID if you read directly to that condition.

This will also need to be accompanied by documentation changes in [the path declaration](https://github.com/edgexfoundry/core-domain/blob/a1ae4c470dcb3a235d101764081e4b3a8c2954c4/src/main/java/org/edgexfoundry/domain/meta/Addressable.java#L50).

Reply:
@trcox Thank you for your comments. I have decided to use `path` variable for two reasons. 

Firstly, there was already an idea to use it to append it to topic. Please consult [these lines](https://github.com/edgexfoundry/core-domain/blob/master/src/main/java/org/edgexfoundry/domain/meta/Addressable.java#L190-L198). 

Secondly, Jim didn't want to add any new field to Addressable. I tried to subclass Addressable, but it provoked a chain reaction of subclassing and I considered this inappropriate, since I didn't want to provoke major changes in code to effectuate minor changes in functionality.

Finally, I've disambigued `path.length() == 1` situation by `path.length() > 1` check. Consequently, I enforce path length to contain at least one informative character.

Reply:
Thanks @darkodraskovic, this resolves my concern on this. Still will wait on Jim's review for this one (he's in Barcelona this week, so it may be a couple of days).